File: Homework3/FunctionModel/Memory.java
public class Memory implements AtomicModel {
    private int value1;
    private int value2;
    
    public Memory(int v1, int v2) {
        value1 = v1;
        value2 = v2;
    }
    
    public String output() {
        return "" + value2;
    }

    public int getNumberOfInputs() {
        return 1;
    }

    public void stateTransition(String[] input) {
        value2 = value1;
        value1 = Integer.parseInt(input[0]);
    }
}

File: Homework6/XORSimulation/Memory.cpp
#include <iostream>
#include <string>
#include <sstream>
#include "Model.h"
#include "Memory.h"
using namespace std;

Memory::Memory(int v1, int v2) {
  value1 = v1;
  value2 = v2;
}

string Memory::output() {
  ostringstream strs;
  strs << value2;
  return strs.str();
}

double Memory::timeAdvance() {
  return 1;
}

void Memory::externalTransition(string inputs[], double timeOfInput) {
  value2 = value1;
  value1 = stoi(inputs[0]);
}

void Memory::internalTransition() {
  //Do nothing
}

void Memory::confluentTransition(string inputs[], double timeOfInput) {
  value2 = value1;
  value1 = stoi(inputs[0]);
}
